README.md

```diff
@@ -55,10 +55,120 @@ console.log(hi); // => howdy world!
 
 # Publishing Binaries
 
-It's a good idea to publish pre-built binaries of your module if you want others to be able to install it properly on their own systems, which can very likely not have a compiler like `gcc` or `clang`. Node-pre-gyp does a lot of the heavy lifting for us (like detecting which system you are building on and deploying to s3) but you'll need a few things configured to get started.
+It's a good idea to publish pre-built binaries of your module if you want others to be able to easily install it on their system without needing to install a compiler like g++ or clang++. Node-pre-gyp does a lot of the heavy lifting for us (like detecting which system you are building on and deploying to s3) but you'll need a few things configured to get started.
+
+1) In the `package.json`, update the `"binary"` field to the appropriate s3 bucket `host`.
+
+For Mapbox staff we recommend using a host setting of `"host": "https://mapbox-node-binary.s3.amazonaws.com",` which will publish to `s3://mapbox-node-binary/<your module name>`.
+
+Note: for namespaced modules the path will end up being `s3://mapbox-node-binary/@org/<your module name>`.
+
+2) Copy the ci.template.js
+
+Copy the `ci.template.js` from this repo into your repo and place it at  `./cloudformation/ci.template.js`
+
+3) Install deps
+
+```
+npm install @mapbox/cfn-config
+npm install @mapbox/cloudfriend
+```
+
+4) Create a user with permissions to upload to `s3://mapbox-node-binary/<your module name>/`
+
+First configure your AWS creds. You will need to set at least the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` keys. And also `AWS_SESSION_TOKEN` if you are using 2-factor auth.
+
+Then run:
+
+```bash
+./node_modules/.bin/validate-template cloudformation/ci.template.js
+./node_modules/.bin/build-template cloudformation/ci.template.js > cloudformation/ci.template
+```
+
+Next we will actually create the user. But first let's discuss what happens here. In addition to creating the user we also write details about the user to a separate bucket (for east auditing purposes). In the below command we:
+
+  - Create a `ci-binary-publish` user
+  - Using the `cloudformation/ci.template`
+  - And ask to save our stack configuration to the `cfn-config` bucket. If you are outside of Mapbox, see [you will need to pass a different bucket and also a `--template-bucket` option](https://github.com/mapbox/cfn-config#prerequisites)
+
+Now, run the command to create the user:
+
+```
+./node_modules/.bin/cfn-config create ci-binary-publish cloudformation/ci.template -c cfn-configs
+```
+
+It will prompt you, choose:
+
+  - New configuration
+  - Ready to create the stack? Y
+
+It will fail if the stack already exists. In this case you can recreate a new user by deleting the stack by running `./node_modules/.bin/cfn-config delete ci-binary-publish cloudformation/ci.template` and then creating a new one.
+
+5) Get the user keys
```

Comment:
You can run `cfn-config info <stack_name>` to get this information as well. In this case, it looks like:

```
~/mapbox/node-cpp-skel[cf]$ cfn-config info ci-binary-publish
{
  "StackId": "arn:aws:cloudformation:us-east-1:......",
  "StackName": "node-cpp-skel-ci-binary-publish",
  "Description": "user for publishing to s3://mapbox-node-binary/@mapbox/node-cpp-skel",
  "Parameters": {},
  "CreationTime": "2017-06-22T20:48:35.469Z",
  "StackStatus": "CREATE_COMPLETE",
  "DisableRollback": false,
  "NotificationARNs": [],
  "Capabilities": [
    "CAPABILITY_IAM",
    "CAPABILITY_NAMED_IAM"
  ],
  "Outputs": {
    "AccessKeyId": "...",
    "SecretAccessKey": "..."
  },
  "Tags": {},
  "Region": "us-east-1"
}
```